acs/coresets.py

```diff
@@ -158,6 +158,23 @@ def _step(self, m, w, **kwargs):
         return w
 
 
+class PBALD(ImportanceSampling):
+
+    def __init_build(self, M=1, seed=None):
+        pass
+
+    def _step(self, m, w, **kwargs):
+        distributions = np.clip(self.scores, 0, None)
```

Comment:
This is not correct.

m is the acquisition batch index not the size.

M is the acquisition batch size.

Please replace with:

```

def _init_build(self, M=1, seed=None):
    distributions = np.clip(self.scores, 0, None)
    distributions /= distributions.sum()

    available_M = (distributions > 0).sum()
    indices = np.random.choice(len(distributions), min(M, available_M), replace=False, p=distributions)
    
    if available_M < m:
        additional_indices = np.random.choice(len(distributions), m - available_M, replace=False, p=(distributions==0.0)/(len(distributions) - available_M)
    indices = np.concat(indices, available_M)
    
    self.indices = indices


def _step(self, m, w, **kwargs):
    w[self.indices[m]] = 1.
```